portfolio/src/main/webapp/comments.js

```diff
@@ -0,0 +1,153 @@
+
+
+/**
+ * Using fetch request content from servlet and add to home page
+ */
+function addMessage(pageNumber = 0, sortMethod = "DESCENDING") {
+  /**
+   * GET does not support additional URL params in the body
+   * So...
+   * A novel hack around this (rather than rewrite the GET logic)
+   * Is to just manually add the strings into the fetch url
+   * This 'works' because the java function getParameter 
+   * Will still parse a manually constructed URL string
+   * So...
+   */
+  const numShown = getCommentLimit();
+  clearMessageDiv();
+  const url = `/data?pageNumber=${pageNumber}&numShown=${numShown}&sortMethod=${sortMethod}`;
+  fetch(url).then(response => response.json()).then((quote) => {
+    const messageContainerDiv = document.getElementById("message-container");
+    quote.forEach(Comment => messageContainerDiv.appendChild(createMessageElements(Comment)));
+  });
+}
+
+/**
+ * Build the elements of an individual message using JSON data
+ * Also create a delete button
+ */
+function createMessageElements(Comment) {
+  var messageDiv = document.createElement("div");
+
+  var commentElement = document.createElement("p");
+  commentElement.innerText = `${Comment.comment} posted at ${Comment.timestamp}`;
+
+  const deleteButtonElement = document.createElement("button");
+  deleteButtonElement.innerText = "Delete comment";
+  deleteButtonElement.addEventListener('click', () => {
+    //remove from datastore and DOM
+    deleteComment(Comment);
+    messageDiv.remove();
+  });
+
+  var likeAndButtonElements = document.createElement("div");
+  var likesElement = document.createElement("p");
+  var likes = 0;
+  likesElement.innerText = likes.toString() + " upvotes";
+
+  var likeButtonElement = document.createElement("button");
+  likeButtonElement.innerText = "Upvote";
+  likeButtonElement.addEventListener('click', () => {
+    //TODO: ADD BACKEND SUPPORT FOR THIS...
+    likes++;
+    likesElement.innerText = likes + " upvotes";
+  });
+
+  likeAndButtonElements.appendChild(likesElement);
+  likeAndButtonElements.appendChild(likeButtonElement);
+  
+  messageDiv.appendChild(commentElement);
+  messageDiv.appendChild(likeAndButtonElements);
+  messageDiv.appendChild(deleteButtonElement);
+  return messageDiv;
+}
+
+function deleteComment(Comment) {
+  const params = new URLSearchParams();
+  params.append('id', Comment.id);
+  fetch('/delete-comment', {method: 'POST', body: params});
+}
+
+function getCommentLimit() {
+  let tmp = (new URL(document.location)).searchParams;
+  let res = tmp.get("numShown");
+  if(!res || res.length == 0){
+    res = "5";
+  }
+  //update the text box to display the amount after refresh 
+  document.getElementById("numShown").value = parseInt(res);
+  return res;
+}
+
+function getSortMethod() {
+  var sortMethodOptions = document.getElementById("sortMethod");
+  var sortMethod = sortMethodOptions.options[sortMethodOptions.selectedIndex].value;
+  document.getElementById("sortMethod").value = sortMethod;
```

Comment:
why are you assigning here?

Reply:
To make sure that on refresh the value is updated to what the user selects

Reply:
Oops!